k8s-scheduler/src/main/java/com/vmware/dcm/ScopedModel.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2018-2020 VMware, Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: BSD-2
+ */
+
+package com.vmware.dcm;
+
+import com.google.common.collect.Sets;
+import com.vmware.dcm.k8s.generated.Tables;
+import org.jooq.Condition;
+import org.jooq.DSLContext;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.SortField;
+import org.jooq.Table;
+import static org.jooq.impl.DSL.field;
+import static org.jooq.impl.DSL.selectFrom;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+import java.util.function.Function;
+
+/**
+ * A Scope to reduce the Model's solver input.
+ *
+ * TODO: dynamically tune the now default parameters {LIMIT_TUNE, CPU_WEIGHT, MEM_WEIGHT}_DEFAULT
+ *       maintain (three) different limitTune variables according to affinity
+ *       add extra limitPriority (provided as a hint)
+ *       capture successful schedules history
+ */
+public class ScopedModel {
+    private static final Logger LOG = LoggerFactory.getLogger(ScopedModel.class);
+    private static final double LIMIT_TUNE_DEFAULT = 1.0;
+    private static final double CPU_WEIGHT_DEFAULT = 0.8;
+    private static final double MEM_WEIGHT_DEFAULT = 0.2;
+
+    private final DSLContext conn;
+    private final Model model;
+
+    ScopedModel(final DSLContext conn, final Model model) {
+        this.conn = conn;
+        this.model = model;
+    }
+
+    /**
+     * Returns a where predicate to filter a given set of nodes.
+     *
+     * @param nodeSet Set of nodes to survive the filtering
+     * @return Where clause predicate
+     */
+    private Condition getWhereClause(final Set<String> nodeSet) {
+        return Tables.SPARE_CAPACITY_PER_NODE.NAME.in(nodeSet);
+    }
+
+    /**
+     * Computes a sorting expression for an ORDER BY clause, based on a sorting strategy.
+     * This sorting strategy sorts according to a weighted sum of nodes' spare cpu and memory.
+     *
+     * @return Sorting expression
+     */
+    private SortField<?> getSorting() {
```

Comment:
When writing methods like this, make sure to refer to a clear noun. So perhaps call this `getOrderByExpression` or `getSortingExpression`.